src/fact_search_and_download.py

```diff
@@ -49,16 +52,20 @@ def main():
     setup_logging(arguments.debug)
     try:
         search_query = get_and_validate_query(arguments.query, arguments.queryfile)
-    except ValueError as e:
+    except json.JSONDecodeError as e:
         sys.exit('Invalid json: {}'.format(e))
     except RuntimeError as e:
         sys.exit('No query given. Use -q or -Q option')
     host = arguments.host
     storage_directory = prepare_storage_dir(arguments.destination)
-    if arguments.firmware:
-        search_result_json = make_search_request_firmware(host, search_query)
-    else:
-        search_result_json = make_search_request_file(host, search_query)
+    try:
+        if arguments.firmware:
+            search_result_json = make_search_request_firmware(host, search_query)
+        else:
+            search_result_json = make_search_request_file(host, search_query)
+    except (requests.RequestException, requests.HTTPError, requests.ConnectionError, json.JSONDecodeError, RuntimeError) as e:
+        sys.exit('Host not reachable: {}'.format(e))
+
     downloaded_files = 0
     logging.info('start download of {} files'.format(len(search_result_json['uids'])))
     for firmware_uid in search_result_json['uids']:
```

Comment:
l. 74 `if downloaded_files % 10 == 0:` will not log the last files (I don't know if this is intended behavior)